list.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require_relative 'my_enumerable'
+
+# Created a new class called List that inherits from MyEnumerable
+class List
+  include MyEnumerable
+
+  def initialize(*list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
```

Comment:
- [ ] For this module, we actually expect you to differently create your own `each` method from scratch instead of using the specified inbuilt array method. It should be similar to the example below:

```ruby
class MyList

[...]

def each
    return to_enum(:each) unless block_given?

    counter = 0
    while counter < @list.length
      yield(@list[counter])
      counter += 1
    end
    @list
end

[...]

end
```